packages/arb-ts/src/lib/l2Bridge.ts

```diff
@@ -118,75 +135,43 @@ export class L2Bridge {
     ](erc20l1Address, to, amount, '0x', overrides)
   }
 
-  public async updateAllL2Tokens(): Promise<Tokens> {
-    for (const l1Address in this.l2Tokens) {
-      const l2Address = this.l2Tokens[l1Address]?.ERC20?.contract.address
-      if (l2Address) {
-        await this.getAndUpdateL2TokenData(l1Address, l2Address)
-      }
-    }
-    return this.l2Tokens
+  public async contractExists(contractAddress: string): Promise<boolean> {
+    const contractCode = await this.l2Provider.getCode(contractAddress)
+    return !(contractCode.length > 2)
   }
 
-  public async getAndUpdateL2TokenData(
+  public async getL2TokenData(
     erc20L1Address: string,
     l2ERC20Address: string
-  ): Promise<L2TokenData | undefined> {
-    const tokenData = this.l2Tokens[erc20L1Address] || {
-      ERC20: undefined,
-      CUSTOM: undefined,
-    }
+  ): Promise<L2TokenData> {
     const walletAddress = await this.getWalletAddress()
-
     // check if standard arb erc20:
-    if (!tokenData.ERC20) {
-      if ((await this.l2Provider.getCode(l2ERC20Address)).length > 2) {
-        const arbERC20TokenContract = await StandardArbERC20__factory.connect(
-          l2ERC20Address,
-          this.l2Signer
-        )
-        const [balance] = await arbERC20TokenContract.functions.balanceOf(
-          walletAddress
-        )
-        tokenData.ERC20 = {
-          contract: arbERC20TokenContract,
-          balance,
-        }
-      } else {
-        console.info(
-          `Corresponding ArbERC20 for ${erc20L1Address} not yet deployed (would be at ${l2ERC20Address})`
-        )
-      }
-    } else {
-      const arbERC20TokenContract = await StandardArbERC20__factory.connect(
-        l2ERC20Address,
-        this.l2Signer
-      )
-      const [balance] = await arbERC20TokenContract.functions.balanceOf(
-        walletAddress
+    if (!this.contractExists(l2ERC20Address))
+      throw new Error(
+        `Corresponding ArbERC20 for ${erc20L1Address} not yet deployed (would be at ${l2ERC20Address})`
```

Comment:
Possibly unnecessary check

Reply:
Though maybe we want it for the clearer error messages?

Reply:
If that's the case we could use a try catch thing where we only check for existence if a call reverts

Reply:
Yea the idea was for clearer error messages, but I don't think its worth it here